File: frontend/src/app/modules/user/models/user.dto.ts
export interface UserDTO {
  userId?: string;
  name?: string | null;
  email?: string | null;
  role?: string | null;
  password?: string | null;
  profileImageUrl?: string | null;
  base64Image?: string | null;
}

export class User implements UserDTO {
  userId?: string;
  name?: string | null = null;
  email?: string | null = null;
  role?: string | null = null;
  password?: string | null = null;
  profileImageUrl?: string | null = null;
  base64Image?: string | null = null;
}

File: backend/00.Persistence/Infrastructure/Dto/UserDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Infrastructure.Dto
{
    public class UserDto
    {
        public Guid UserId { get; set; }
        public string Name { get; set; } = "";
        public string Email { get; set; } = "";
        public string Role { get; set; } = "";
        public string PasswordHash { get; set; } = "";
        public string? ProfileImageUrl { get; set; }
        public string? Base64Image { get; set; }
    }
}
